agent/engine/docker_task_engine.go

```diff
@@ -597,6 +612,20 @@ func (engine *DockerTaskEngine) createContainer(task *api.Task, container *api.C
 	seelog.Infof("Created container name mapping for task %s - %s -> %s", task, container, containerName)
 	engine.saver.ForceSave()
 
+	// Create metadata directory and file then populate it with common metadata of all containers of this task
+	// Afterwards add this directory to the container's mounts if file creation was successful
+	// CreateMetadata has a call to docker API for the client version so this may be block our container creation
+	// But we require the host Config binds to be properly set before we create the container
+	if engine.cfg.ContainerMetadataEnabled {
```

Comment:
I am not sure if I can move all of this to its own method. Since HostConfig.Binds is a shared variable throughout the task engine, passing it by reference and modifying it directly is risks concurrent modification. I am told passing this by value and reassigning it is a bit safer, although this issue is probably still present.